### DISCUSSION TITLE
Continuous prop not working, beacon still shows. Passed disableBeacon but overlay disappears.

### DISCUSSION THREAD

**Discussion by cantuus** [gilbarbara/react-joyride#685] [Q&A] [answered]
I want to be able to continue through the Product Tour without having to click on the beacons.  I passed `continuous` into the Joyride component but the beacon still shows. I also passed `disableBeacon` into each Step.  It removes the beacon but it also removes the overlay.  I replicated the same properties from the React-Joyride demo but still doesn't seem to work. I double checked the debugger and the properties are set correctly. This was an already existing issue when I first implemented the Joyride component before the component became controlled. Was wondering if this was a bug or some way to proceed through the Tour using the 'Next' button while also keeping the overlay.

**Comment by gilbarbara** [comment, marked as answer]
If the user closes the step and there are more steps to complete, how would the user know about them if not with a beacon?
Do you want to skip the rest of the steps?

You can listen for the callback and stop the tour when the user closes a tooltip or use custom components to remove the functionality that you don't want.

**Reply by cantuus** [reply]
I want the Product Tour to keep proceeding through the steps by clicking on a Next button in the tooltip without having to click on a beacon prior to rendering the tooltip.  So to illustrate, the first tooltip automatically pops up when the DOM renders, a user clicks on the next button in the tooltip. Then it scrolls to the next target and automatically brings up the tooltip. This is repeated for the rest of the steps.

This behavior is doable if you pass disableBeacon into each step but the joyride overlay does not render in the DOM.

Do you have any suggestions on which properties to look into in the callback to replicate this behavior?

**Reply by fmatheus** [reply]
I have the same problem here, continuous is been ignored after update react-joyride from 2.2.1 to 2.3.0 (downgrading workaround the issue).

**Reply by Stemonitis** [reply]
I have the exact same problem, the beacons reappear after each step instead of continuous flow of tips. Did anyone find a solution?

**Reply by gilbarbara** [reply]
@Stemonitis Please post a minimal reproduction in codesandbox

**Reply by Stemonitis** [reply]
> @Stemonitis Please post a minimal reproduction in codesandbox
I spent some time trying to reproduce the problem, but for some reason the exact same code does not have this issue in the sandbox, but does have it in the browser (I use parcel as the package manager). At this point I have absolutely no idea what is it that makes it behave that way, but if you have any clues, please, let me know. Thanks!

**Comment by gilbarbara** [comment]
So, do you want the same behavior as the [demo](https://react-joyride.com/)? Did you check the demo's [source code](https://github.com/gilbarbara/react-joyride-demo)?

Anyway, I really can't say what's going on without seeing some code...
Please create a minimal reproducible example in codesandbox

**Reply by cantuus** [reply]
I replicated the same props for my current product tour in my project but still come across the same issue.  

However I think it might have to do more with the project itself.  I made a simple CodeSandBox and it works fine. There might be some underlying problems that might be affecting why the overlay disappears after passing `disableBeacon` into each step and why the beacon still shows after passing `continuous` into Joyride. I will have to continue debugging on my end.

https://codesandbox.io/s/trusting-drake-cm88r?file=/src/App.js

**Comment by hungphannapaglobal** [comment]
i have the same issue, although i was copy code on your [demo](https://codesandbox.io/s/trusting-drake-cm88r) and have the same version, but `continuous` still  not working

**Reply by hungphannapaglobal** [reply]
i have to downgrade to 2.2.1 and it work again
